lib/fizzy/execute.cpp

```diff
@@ -631,10 +628,11 @@ std::unique_ptr<Instance> instantiate(Module module,
 execution_result execute(
     Instance& instance, FuncIdx func_idx, std::vector<uint64_t> args, int depth)
 {
-    assert(depth >= 0 && depth <= CallStackLimit);
+    if (depth < 0 || depth > CallStackLimit)
```

Comment:
Too defensive IMHO. Move `depth < 0` back to assert.
Note: I don't consider `execute()` a public API.

Reply:
I changed it to if, because now via `ExternalFunction` returned from `find_exported_function` it can be called with arbitrary `depth` value. Somehow this feels more like a value like can come from outside of the library.
But I guess it's still not public API, I can change it.